x/feegrant/internal/types/tx.go

```diff
@@ -0,0 +1,241 @@
+package types
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/tendermint/tendermint/crypto"
+	"github.com/tendermint/tendermint/crypto/multisig"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth/exported"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+)
+
+var (
+	_ sdk.Tx = FeeGrantTx{}
+
+	maxGasWanted = uint64((1 << 63) - 1)
+)
+
+// FeeGrantTx wraps a Msg with Fee and Signatures,
+// adding the ability to delegate the fee payment
+// NOTE: the first signature responsible for paying fees, either directly,
+// or must be authorized to spend from the provided Fee.FeeAccount
+type FeeGrantTx struct {
```

Comment:
question/non-blocking: Is there absolutely no way for us to utilize the existing `StdTx` type (it's ok to change the `Fee` type)?

Reply:
This was your idea: https://github.com/cosmos/cosmos-sdk/pull/5207#issuecomment-543340544

I proposed to make the change to StdTx/StdFee

Reply:
So how should we proceed on this @alexanderbez? We can either keep the custom tx or just add a `FeeAccount sdk.AccAddress` field to `StdTx` as was the original design.

Reply:
Upon reflecting on this, my thought is that if this is in the SDK and going onto cosmos-hub, then to me it makes sense this would be part of `StdTx`. If this were only going on a zone like Regen, we should have a custom tx